protocol-units/execution/executor/src/v1.rs

```diff
@@ -1,34 +1,49 @@
-// FIXME: glob imports are bad style
-use crate::*;
-use aptos_types::transaction::SignedTransaction;
-use maptos_opt_executor::Executor;
+use crate::{BlockMetadata, ExecutableBlock, Executor, HashValue, SignedTransaction};
+use aptos_api::runtime::Apis;
+use maptos_fin_view::FinalityView;
+use maptos_opt_executor::Executor as OptExecutor;
+use maptos_execution_util::config::Config;
 use movement_types::BlockCommitment;
 
 use async_channel::Sender;
+use async_trait::async_trait;
 use tracing::debug;
 
 #[derive(Clone)]
-pub struct MonzaExecutorV1 {
-	// this rwlock may be somewhat redundant
-	pub executor: Executor,
+pub struct ExecutorV1 {
+	pub executor: OptExecutor,
+	finality_view: FinalityView,
 	pub transaction_channel: Sender<SignedTransaction>,
 }
 
-impl MonzaExecutorV1 {
-	pub fn new(executor: Executor, transaction_channel: Sender<SignedTransaction>) -> Self {
-		Self { executor, transaction_channel }
+impl ExecutorV1 {
+	pub fn new(executor: OptExecutor, finality_view: FinalityView, transaction_channel: Sender<SignedTransaction>) -> Self {
+		Self { executor, finality_view, transaction_channel }
 	}
 
-	pub async fn try_from_env(
+	pub fn try_from_env(
 		transaction_channel: Sender<SignedTransaction>,
 	) -> Result<Self, anyhow::Error> {
-		let executor = Executor::try_from_env()?;
-		Ok(Self::new(executor, transaction_channel))
+		let config = Config::try_from_env()?;
+		Self::try_from_config(transaction_channel, &config)
+	}
+
+	pub fn try_from_config(
+		transaction_channel: Sender<SignedTransaction>,
+		config: &Config,
+	) -> Result<Self, anyhow::Error> {
+		let executor = OptExecutor::try_from_config(&config.aptos)?;
+		let finality_view = FinalityView::try_from_config(
+			executor.db.reader.clone(),
+			executor.mempool_client_sender.clone(),
+			&config.aptos,
+		)?;
+		Ok(Self::new(executor, finality_view, transaction_channel))
 	}
 }
 
-#[tonic::async_trait]
-impl MonzaExecutor for MonzaExecutorV1 {
+#[async_trait]
+impl Executor for ExecutorV1 {
```

Comment:
There's the `Executor` trait and this implementation type. I suppose the trait is useful for creating mocks, otherwise I'd just have inherent public methods on the type. I don't like naming traits with the `Trait` suffix (looking at you, Aptos), but we'll need to have a couple of names, otherwise having both named `Executor` in one crate may be too confusing.